mysql/mysql_test.go

```diff
@@ -180,6 +180,8 @@ func (t *mysqlTestSuite) TestMysqlGTIDMinus(c *check.C) {
 		{"3E11FA47-71CA-11E1-9E33-C80AA9429562:20-57:60-90", "3E11FA47-71CA-11E1-9E33-C80AA9429562:23", "3E11FA47-71CA-11E1-9E33-C80AA9429562:20-22:24-57:60-90"},
 		{"3E11FA47-71CA-11E1-9E33-C80AA9429562:20-57:60-90", "3E11FA47-71CA-11E1-9E33-C80AA9429562:22-70", "3E11FA47-71CA-11E1-9E33-C80AA9429562:20-21:71-90"},
 		{"3E11FA47-71CA-11E1-9E33-C80AA9429562:28-57", "3E11FA47-71CA-11E1-9E33-C80AA9429562:28-57", ""},
```

Comment:
found a failed case 
```
		{"3E11FA47-71CA-11E1-9E33-C80AA9429562:20-21", "3E11FA47-71CA-11E1-9E33-C80AA9429562:21", "3E11FA47-71CA-11E1-9E33-C80AA9429562:20"},

```

please check it

Reply:
You right.

New version with changes:
 * merge 2 loops into one, so it will be easier to reason about
 * explicit IF conditions to make it clear which cases are handled in each branch

Hope, this time it is correct for all inputs